### ISSUE TITLE
[Enhancement] Spotlight & Time-Machine Commands

### ISSUE THREAD

**Issue by petermbenjamin** [rgcr/m-cli#96]
Found this [Apple OS X Command Line documentation](http://ss64.com/osx/), which could be a gold-mine of potential additions to this project.

### [Spotlight Search](http://ss64.com/osx/mdfind.html)

```bash
$ mdfind
Usage: mdfind [-live] [-count] [-onlyin directory] [-name fileName | -s smartFolderName | query]
list the files matching the query
query can be an expression or a sequence of words

	-attr <attr>      Fetches the value of the specified attribute
	-count            Query only reports matching items count
	-onlyin <dir>     Search only within given directory
	-live             Query should stay active
	-name <name>      Search on file name only
	-reprint          Reprint results on live update
	-s <name>         Show contents of smart folder <name>
	-0                Use NUL (``\0'') as a path separator, for use with xargs -0.

example:  mdfind image
example:  mdfind -onlyin ~ image
example:  mdfind -name stdlib.h
example:  mdfind "kMDItemAuthor == '*MyFavoriteAuthor*'"
example:  mdfind -live MyFavoriteAuthor
```

### ~~[App Store Software Updates](http://ss64.com/osx/softwareupdate.html)~~ Already implemented
```bash
$ softwareupdate
usage: softwareupdate <cmd> [<args> ...]

** Catalog Management:
	--set-catalog <URL>	Set the new catalog URL (requires privileges)
	--clear-catalog		Clear the catalog URL back to defaults (requires privileges)

** Manage Updates:
	-l | --list		List all appropriate update labels (options:  --no-scan)
	-d | --download		Download Only
	-e | --cancel-download		Cancel a download
	-i | --install		Install
		<label> ...	specific updates
		-a | --all		All appropriate updates
		-r | --recommended	Only recommended updates
	--background		Trigger a background scan and update operation
	--ignore <label> ...	Ignore specific updates
	--reset-ignored		Clear all ignored updates

** Other Tools:
	--suspend-background 	Suspend background operations from occurring temporarily (use --duration to specify duration to suspend in seconds)
		--duration <duration>)		Optional duration in seconds to suspend background operations (defaults to 5*60 seconds)
	--dump-state		Log the internal state of the SU daemon to /var/log/install.log
	--evaluate-products		Evaluate a list of product keys specified by the --products option

** Options:
	--no-scan		Do not scan when listing or installing updates (use available updates previously scanned)
	--products		A comma-separated (no spaces) list of product keys to operate on.
	--force			Force an operation to complete.  Use with --background to trigger a background scan regardless of "Automatically check" pref

--verbose	Enable verbose output
--help	Print this help
```

### [Time Machine](http://ss64.com/osx/tmutil.html)
```bash
$ tmutil
Usage: tmutil version

Usage: tmutil enable

Usage: tmutil disable

Usage: tmutil startbackup [-a | --auto] [-b | --block] [-r | --rotation] [-d | --destination dest_id]

Usage: tmutil stopbackup

Usage: tmutil enablelocal

Usage: tmutil disablelocal

Usage: tmutil snapshot

Usage: tmutil delete snapshot_path ...

Usage: tmutil restore [-v] src ... dst

Usage: tmutil compare [-@acdefghlmnstuEX] [-D depth] [-I name]
       tmutil compare [-@acdefghlmnstuEX] [-D depth] [-I name] snapshot_path
       tmutil compare [-@acdefghlmnstuEUX] [-D depth] [-I name] path1 path2

Usage: tmutil setdestination [-a]  mount_point
       tmutil setdestination [-ap] afp://user[:pass]@host/share

Usage: tmutil removedestination destination_id

Usage: tmutil destinationinfo [-X]

Usage: tmutil addexclusion [-p|-v] item ...

Usage: tmutil removeexclusion [-p|-v] item ...

Usage: tmutil isexcluded item ...

Usage: tmutil inheritbackup machine_directory
       tmutil inheritbackup sparse_bundle

Usage: tmutil associatedisk [-a] mount_point volume_backup_directory

Usage: tmutil latestbackup

Usage: tmutil listbackups

Usage: tmutil machinedirectory

Usage: tmutil calculatedrift machine_directory

Usage: tmutil uniquesize path ...

Usage: tmutil verifychecksums path ...
```

**Comment by rgcr** [comment]
Hi @pmbenjamin first of all thank you so much for your ideas and contributions.
`softwareupdate` has already a basic implemented in the `update` plugin, you can add more functionality to it if you want.

It's a good idea to have plugins for `tmutil` and `mdfind`. Feel free to create the proper PR's 👍

**Comment by thegranddesign** [comment]
Some of the time machine commands are implemented in #102

**Comment by petermbenjamin** [comment]
Since Time Machine commands are implemented in #102 and `softwareupdate` is basically implemented in the update plugin, I am closing this issue.
